File: src/mod8_TicTacToe/java/controller/GameController.java
package mod8_TicTacToe.java.controller;

import mod8_TicTacToe.java.model.Board;
import mod8_TicTacToe.java.model.Player;
import mod8_TicTacToe.java.model.Symbol;

import java.util.Scanner;

public class GameController {
    Player player1;
    Player player2;
    Board board;

    {
        initPlayers();
    }

    public GameController() {
        board = new Board();
    }

    public void playGame() {
        Scanner in = new Scanner(System.in);
        Player currentPlayer = player1;
        while (true) {
            board.showCurrentState();
            System.out.println("Player " + currentPlayer.getName() + " enter your move(row and column)!");
            int row = 0, column = 0;
            System.out.println("row=");
            row = in.nextInt();
            System.out.println("column=");
            column = in.nextInt();
            row--;
            column--;

            if (board.isCellEmpty(row, column)) {
                board.setCell(row, column, currentPlayer.getSymbol());
                board.showCurrentState();
            }
            if (board.isWinner(currentPlayer)) {
                System.out.println("The player " + currentPlayer.getName() + " is win! GAME OVER!");
                break;
            } else {
                currentPlayer = (currentPlayer == player1 ? player2 : player1);
            }
        }
    }

    private void initPlayers() {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter the name of Player 1: ");
        String player1Name = in.nextLine();
        do {
            System.out.println("Choose a symbol for Player " + player1Name + " (X or 0) : ");
            String symbol = in.nextLine();
            if (symbol.length() == 1) {
                if (symbol.equalsIgnoreCase("X") || symbol.equalsIgnoreCase("Х")) {
                    player1 = new Player(player1Name, Symbol.X);
                } else {
                    player1 = new Player(player1Name, Symbol.O);
                }
            }
        } while (player1 == null);

        System.out.println("Enter the name of Player 2: ");
        String player2Name = in.nextLine();
        if (player1.getSymbol() == Symbol.X) {
            player2 = new Player(player2Name, Symbol.O);
        } else {
            player2 = new Player(player2Name, Symbol.X);
        }
//        in.close();
    }
}

File: src/mod8_TicTacToe/kotlin/controller/GameController.kt
package mod8_TicTacToe.kotlin.controller

import mod8_TicTacToe.kotlin.model.Board
import mod8_TicTacToe.kotlin.model.Player
import mod8_TicTacToe.kotlin.model.Symbol

class GameController {

    private lateinit var player1: Player
    private lateinit var player2: Player
    private val board = Board()

    init {
        initPlayer()
    }

    fun playGame(){
        var currentPlayer=player1
        while (true){
            board.showCurrentState()
            println("Player ${currentPlayer.name}, enter your move(row and column)!")
            var row = 0
            var column = 0
            println("row=")
            row = readLine()!!.toInt() -1
            println("column=")
            column =  readLine()!!.toInt() -1
            if (board.isCellEmpty(row, column)) {
                board.setCell(row, column, currentPlayer.symbol)
                board.showCurrentState()
            }
            if (board.isWinner(currentPlayer)) {
                println("The player ${currentPlayer.name} is win! GAME OVER!")
                break
            } else {
                currentPlayer = (if (currentPlayer === player1) player2 else player1)
            }
        }
    }


    private fun initPlayer() {
        println("Enter the name of Player 1: ")
        val namePlayer1 = readln()
        println("Choose a symbol for Player $namePlayer1 (X or O): ")
        val symbolfromConsole = readln().uppercase()
        val symbol = when (symbolfromConsole) {
            "X", "Х" -> Symbol.X
            else -> Symbol.O
        }
        player1=Player(namePlayer1,symbol)
        println("Enter the name of Player 2: ")
        val namePlayer2 = readln()
        player2=Player(namePlayer2,if (player1.symbol==Symbol.X) Symbol.O else Symbol.X)


    }
}